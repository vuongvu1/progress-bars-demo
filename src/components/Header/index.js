import React from 'react';
import styled from 'styled-components';
import { NavLink } from 'react-router-dom';

// #region Styles
const Wrapper = styled.header`
  background-color: ${(props) => props.theme.black};
  color: ${(props) => props.theme.white};
  position: fixed;
  z-index: 1;
  width: 100%;
  top: 0;
  left: 0;
  padding: 0 20px;
  height: 50px;

  ul {
    width: 100%;
    margin: 0;
    padding: 0;
    display: flex;
    list-style-type: none;
    user-select: none;
    height: 100%;

    > li {
      display: flex;
      justify-content: flex-start;
      align-items: center;

      a {
        text-decoration: none;
      }

      &:first-child {
        flex: 0 0 auto;
        width: 20%;
        font-weight: bold;
        a {
          color: ${(props) => props.theme.darkBlue};
        }
      }
      &:nth-child(2) {
        flex: 1 1 auto;
        a {
          color: ${(props) => props.theme.white};
          margin-right: 20px;

          &:hover {
            color: ${(props) => props.theme.darkBlue};
          }
        }
      }
      &:last-child {
        flex: 0 0 auto;
      }
    }
  }
`;
// #endregion Styles

const Header = () => (
  <Wrapper data-cy='header'>
    <ul>
      <li>
        <NavLink exact to='/'>
          <span role='img' aria-label='Star'>‚≠ê</span> Home
        </NavLink>
      </li>
      <li>
        <NavLink exact to='/progress-bars-demo'>Progress Bars Demo</NavLink>
        <NavLink exact to='/error'>Error Page</NavLink>
      </li>
      <li>Hi, Anonymous</li>
    </ul>
  </Wrapper>
);

export default Header;
