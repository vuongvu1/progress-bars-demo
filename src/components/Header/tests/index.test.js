import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

import Header from '../index';

const renderComponent = (props = {}) => render(
  <BrowserRouter>
    <Header {...props} />,
  </BrowserRouter>,
);

describe('<Header />', () => {
  it('should render <Header />', () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });
});
