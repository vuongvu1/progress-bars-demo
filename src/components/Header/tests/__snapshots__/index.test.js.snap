// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render <Header /> 1`] = `
<div>
  <header
    class="sc-bdVaJa dGTFIM"
    data-cy="header"
  >
    <ul>
      <li>
        <a
          aria-current="page"
          class="active"
          href="/"
        >
          <span
            aria-label="Star"
            role="img"
          >
            ‚≠ê
          </span>
           Home
        </a>
      </li>
      <li>
        <a
          href="/progress-bars-demo"
        >
          Progress Bars Demo
        </a>
        <a
          href="/error"
        >
          Error Page
        </a>
      </li>
      <li>
        Hi, Anonymous
      </li>
    </ul>
  </header>
  ,
</div>
`;
